packages/fx-core/src/plugins/resource/bot/v2/index.ts

```diff
@@ -65,21 +66,26 @@ export class BotPluginV2 implements ResourcePlugin {
   }
 
   async scaffoldSourceCode(ctx: Context, inputs: Inputs): Promise<Result<Void, FxError>> {
-    return await scaffoldSourceCodeAdapter(ctx, inputs, this.plugin);
+    return catchAndThrow(() => this.impl.scaffoldSourceCode(ctx, inputs));
+    // return await scaffoldSourceCodeAdapter(ctx, inputs, this.plugin);
   }
 
   async generateResourceTemplate(
     ctx: Context,
     inputs: Inputs
   ): Promise<Result<ResourceTemplate, FxError>> {
-    return await generateResourceTemplateAdapter(ctx, inputs, this.plugin);
+    return catchAndThrow(() => this.impl.generateResourceTemplate(ctx, inputs));
+    // return await generateResourceTemplateAdapter(ctx, inputs, this.plugin);
   }
+
   async updateResourceTemplate(
     ctx: Context,
     inputs: Inputs
   ): Promise<Result<v2.ResourceTemplate, FxError>> {
+    // return catchAndThrow(() => this.impl.updateResourceTemplate(ctx, inputs));
```

Comment:
I am a little curious on why scaffold/genBicep is using catchAndThrow(), but not for updateBicep?

Reply:
Because updateBicep is not implemented yet.